app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,133 @@
 class SendgridApi
-  extend SingleForwardable
+  class Error < StandardError; end
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  TIMEOUT = 2 # seconds
+
+  def initialize
+    @connection = Excon.new(Rails.configuration.sendgrid_api_host,
+      persistent: true,
+      connect_timeout: TIMEOUT)
+  end
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    route = 'spamreports.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def bounced?(email)
-    api { bounces.retrieve(email: email).any? }
+    route = 'bounces.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
   end
 
   def remove_from_bounce_list(email)
-    api { bounces.delete(email: email) }
+    route = 'bounces.delete.json'
+    body = { email: email }
+
+    response = post_request(route, body)
+
+    check_email_error(response)
+    true
+  rescue => e
+    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    false
```

Comment:
Yes, not keen on the use of errors as flow control. With this PR I wanted to use the existing specs as much as possible to be sure that I didn't miss anything and to keep the number of changes small.

Ideally I'd like to pull out the API calls to a separate class so that SendgridApi uses that and makes the decisions. Then if later on there is a good open source Sendgrid client the implementation of SendgridApi can be changed to use it